uaclient/contract.py

```diff
@@ -194,11 +218,41 @@ def _request_machine_token_update(
         @param detach: Boolean set True if detaching this machine from the
             active contract. Default is False.
 
+        @return: Dict of the JSON response containing refreshed machine-token
+        """
+        data = self._get_platform_data(machine_id)
+        response = self._get_updated_machine_token_details(
+            data, machine_token, contract_id, detach
+        )
+        if not detach:
+            self.cfg.write_cache("machine-token", response)
+            util.get_machine_id.cache_clear()
+            machine_id = response.get("machineTokenInfo", {}).get(
+                "machineId", data.get("machineId")
+            )
+            self.cfg.write_cache("machine-id", machine_id)
+        return response
+
+    def _get_updated_machine_token_details(
+        self,
+        data: Dict,
+        machine_token: str,
+        contract_id: str,
+        detach: bool = False,
+    ) -> Dict[str, Any]:
+        """Get the updated machine token from the contract server.
+
+        @param data: specific machine platform data
+        @param machine_token: The machine token needed to talk to
+            this contract service endpoint.
+        @param contract_id: Unique contract id provided by contract service.
+        @param detach: Boolean set True if detaching this machine from the
+            active contract. Default is False.
+
```

Comment:
I'm not sure we need this refactor anymore since we created a separate function for the GET request